plonky2/examples/factorial.rs

```diff
@@ -0,0 +1,35 @@
+use anyhow::Result;
+use plonky2::field::types::Field;
+use plonky2::iop::witness::PartialWitness;
+use plonky2::plonk::circuit_builder::CircuitBuilder;
+use plonky2::plonk::circuit_data::CircuitConfig;
+use plonky2::plonk::config::{GenericConfig, PoseidonGoldilocksConfig};
+
+fn main() -> Result<()> {
+    const D: usize = 2;
+    type C = PoseidonGoldilocksConfig;
+    type F = <C as GenericConfig<D>>::F;
+
+    let config = CircuitConfig::standard_recursion_config();
+
+    let pw = PartialWitness::new();
+    let mut builder = CircuitBuilder::<F, D>::new(config);
+
+    let mut cur_target = builder.one();
+    for i in 2..101 {
+        let i_target = builder.constant(F::from_canonical_u32(i));
+        cur_target = builder.mul(cur_target, i_target);
+    }
```

Comment:
Same as for the Fibonacci example, this will only produce `ConstantGate`s.

I'm not sure how to fix this here, one way would be to instead prove "I know the product `n*(n+1)*...(n+k)`" and have `n` be a public input and `k` be a constant (like 100 here).